sdk/core/core-lro/samples-dev/customPoller.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary demnostrates how to create a custom poller.
+ */
+
+import type { OperationState, OperationStatus, PollerLike } from "@azure/core-lro";
+import { delay } from "@azure/core-util";
+
+const DEFAULT_POLL_INTERVAL_IN_MS = 2000;
+
+type OperationConfig = { id: string };
+
+type RestorableOperationState<TResult, TState extends OperationState<TResult>> = TState & {
+  config: OperationConfig;
+};
+
+async function initOperation<T>(): Promise<RestorableOperationState<T, OperationState<T>>> {
+  // starts the operation
+  return { status: "running", config: { id: "1" } };
+}
+
+async function pollOperation<T>({
+  setDelay,
+  state,
+  options,
+}: {
+  state: RestorableOperationState<T, OperationState<T>>;
+  setDelay: (interval: number) => void;
+  options?: { abortSignal?: AbortSignal };
+}): Promise<void> {
+  if (options?.abortSignal?.aborted) {
+    throw new Error("aborted");
+  }
+  // polls the operation using id while respecting the abort signal in options
+  const { id } = state.config;
+  const response = { id, status: "succeeded", retryAfter: 2000 };
+  // update the state based on the response
+  if (response.status === "succeeded") {
+    state.status = "succeeded";
+    state.result = response as T;
+  } else if (response.status === "failed") {
+    state.status = "failed";
+    state.error = (response as any).error;
+  } else if (response.status === "canceled") {
+    state.status = "canceled";
+  } else {
+    setDelay(response.retryAfter); // updates the delay if the service suggests a different interval
+  }
+}
+
+function deserializeState<T, TState extends OperationState<T>>(
+  serializedState: string,
+): RestorableOperationState<T, TState> {
+  try {
+    return JSON.parse(serializedState).state;
+  } catch (e) {
+    throw new Error(`Unable to deserialize input state: ${serializedState}`);
+  }
+}
+
+function createPoller<T>({
```

Comment:
do we have a plan to put this in src directly? i feel like our current implementation is quite coupled with guidelines so any non-standard LRO would re-write the whole poller.

Reply:
We already have an un-exported definition for an abstract poller in `src` in https://github.com/Azure/azure-sdk-for-js/blob/ef1cfa3d5147479b997e52322b8e09e5c373914f/sdk/core/core-lro/src/poller/poller.ts#L19-L24

I am open to export it if we get enough interest to do so.